test/libponyc/literal_inference.cc

```diff
@@ -98,7 +117,57 @@ TEST_F(LiteralTest, CantInferLitExpr)
 }
 
 
-TEST_F(LiteralTest, CantInferFloatLiteralFromIntType)
+TEST_F(LiteralTest, CantInfer_Op_IntAndFloat )
+{
+  const char* src =
+    "class Foo10b\n"
+    "  fun test(): I32 => 7 and 16.0\n";
+
+  TEST_ERROR(src);
+}
+
+
+TEST_F(LiteralTest, CantInfer_Op_FloatAndInt )
+{
+  const char* src =
+    "class Foo11b\n"
+    "  fun test(): I32 => 7.0 and 16\n";
+
+  TEST_ERROR(src);
+}
+
+
+TEST_F(LiteralTest, CantInfer_Op_FloatAndInt_Float )
+{
+  const char* src =
+    "class Foo12b\n"
+    "  fun test(): F32 => 7.0 and 16\n";
+
+  TEST_ERROR(src);
+}
+
+
+TEST_F(LiteralTest, CantInfer_Op_IntAndFloat_Float )
+{
+  const char* src =
+    "class Foo13b\n"
+    "  fun test(): F32 => 7 and 16.0\n";
+
+  TEST_ERROR(src);
+}
+
+
+TEST_F(LiteralTest, CantInfer_Op_Equals )
+{
+  const char* src =
+    "class Foo14b\n"
+    "  fun test(): Bool => 34 == 79\n";
+
+  TEST_ERROR(src);
+}
+
+
+TEST_F(LiteralTest, CantInfer_Let_FloatLiteralFromIntType)
```

Comment:
In these new names, what is the determining factor for when two words are separated by `_` and when they aren't?